plutus-tx/src/PlutusTx/Blueprint/Definition/Unroll.hs

```diff
@@ -89,8 +89,12 @@ type family Unroll (p :: Type) :: [Type] where
   Unroll BuiltinData = '[BuiltinData]
   Unroll BuiltinUnit = '[BuiltinUnit]
   Unroll BuiltinString = '[BuiltinString]
-  Unroll (BuiltinList a) = Prepend (BuiltinList a) (GUnroll (Rep a))
+  Unroll (BuiltinList a) = Unroll a
+  Unroll (BuiltinPair a b) = Unroll a ++ Unroll b
   Unroll BuiltinByteString = '[BuiltinByteString]
+  Unroll [a] = Unroll a
+  Unroll (a, b) = Unroll a ++ Unroll b
+  Unroll (Maybe a) = Unroll a
```

Comment:
Well I still have no idea why this is desired rather than the `Prepend` thing. I do remember our discussion about lists, pairs etc being an inherent part of a schema, but what if it's a list of lists, is that now "user-space"? And I'm failing to understand how this logic maps onto the `#list`, `#pair` etc built-in types as per the [CIP](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0057).

Sorry for being useless! Maybe the reviewers of the original blueprint PR would know better, CC'ing @kwxm @zliu41.